source/MaterialXRuntime/RtValue.h

```diff
@@ -288,143 +276,276 @@ class RtTypedValue
     /// Return bool value.
     const bool& asBool() const
     {
+#ifndef NDEBUG
+        if (_type != RtType::BOOLEAN)
```

Comment:
Make sure the referenced type is matching the allocated type. Only in debug builds.